pkg/systemd/reporting.jsx

```diff
@@ -116,114 +104,139 @@ class FAFWorkflowRow extends React.Component {
                         console.error(cockpit.format("reporter-ureport was killed with signal $0", exception.exit_signal));
                     }
                 })
-                .finally(() => this.updateStatusFromBus());
+                .finally(() => updateStatusFromBus());
+
+        setProblemState(ProblemState.REPORTING);
+        setProcess(process);
+    };
+
+    return <WorkflowRow label={_("Report to ABRT Analytics")}
+                        message={message}
+                        onCancelButtonClick={onCancelButtonClick}
+                        onReportButtonClick={onReportButtonClick}
+                        problemState={problemState}
+                        reportLinks={reportLinks}
+    />;
+};
+
+const BusWorkflowRow = ({ problem, client, workflow }) => {
+    const [message, setMessage] = React.useState("");
+    const [problemState, setProblemState] = React.useState(ProblemState.REPORTABLE);
+    const [reportLinks, setReportLinks] = React.useState([]);
+    const [task, setTask] = React.useState(null);
+    const label = workflow[1];
+    const inputRef = React.createRef();
+
+    const updateStatusFromBus = () => {
+        const on_get_properties = properties => {
+            if (!properties[0].CanBeReported.v) {
+                setProblemState(ProblemState.UNREPORTABLE);
+                return;
+            }
 
-        this.setState({
-            problemState: ProblemState.REPORTING,
-            process,
-        });
-    }
+            const reportLinks = [];
+            let reported = false;
 
-    render() {
-        return <WorkflowRow label={_("Report to ABRT Analytics")}
-                            message={this.state.message}
-                            onCancelButtonClick={this._onCancelButtonClick}
-                            onReportButtonClick={this._onReportButtonClick}
-                            problemState={this.state.problemState}
-                            reportLinks={this.state.reportLinks}
-        />;
-    }
-}
+            for (const report of properties[0].Reports.v) {
+                if (!("WORKFLOW" in report[1])) {
+                    continue;
+                }
+                if (workflow[0] !== report[1].WORKFLOW.v.v) {
+                    continue;
+                }
+                if (report[0] === "ABRT Server" || report[0] === "uReport") {
+                    continue;
+                }
+                if ("URL" in report[1]) {
+                    reportLinks.push(report[1].URL.v.v);
+                }
+
+                reported = true;
+            }
 
-class BusWorkflowRow extends React.Component {
-    constructor(props) {
-        super(props);
+            if (reported) {
+                setProblemState(ProblemState.REPORTED);
+                setReportLinks(reportLinks);
+            }
+        };
+        const on_get_properties_rejected = exception => {
+            setProblemState(ProblemState.UNREPORTABLE);
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19403-20231002-082424-b39d5ff6-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)